lib/pg_ha_migrations/safe_statements.rb

```diff
@@ -170,6 +175,87 @@ def safe_remove_concurrent_index(table, options={})
     unsafe_remove_index(table, **options.merge(:algorithm => :concurrently))
   end
 
+  # TODO: figure out what options we actually want to allow
+  #
+  # we definitely don't want algorithm and unique
+  def safe_add_concurrent_partitioned_index(table, columns, name_suffix:, if_not_exists: false, using: nil)
+    raise ArgumentError, "Expected <name_suffix> to be present" unless name_suffix.present?
```

Comment:
We need to lookup the table schema to accurately run discovery queries (looking up child partitions / checking index validity), and therefore always use the fully qualified table name when calling `add_index`. Rails will generate a name like `public.foo_column_idx` if the index name isn't explicitly provided, which isn't expected if you call this method as `safe_add_concurrent_partitioned_index :foo, :column`. Additionally, we need the name of the child indexes when calling `ALTER INDEX ... ATTACH PARTITION` (`add_index` returns a `PG::Result` object so we probably _could_ get the index name dynamically, but API doesn't explicitly call out the return type for this method so I don't think we should rely on that).

With all that being said, to simplify this code I think it's perfectly reasonable to just require `name_suffix` to be present

Reply:
Addressed in https://github.com/braintree/pg_ha_migrations/pull/85/commits/71c773df9e90014afd2c4ae82c3c253ffa04fcc1

Reply:
Do we want name suffix? Or just name, use name for the index on the parent, and do whatever we want for the child tables?

One other thing: I think it's OK to require the name to match for the `if_not_exists` (it saves a lot of effort on trying to lookup indexes and match them opportunistically), but I wonder if we should explicitly document as such.

Reply:
> Do we want name suffix? Or just name, use name for the index on the parent, and do whatever we want for the child tables?

That's a really good point because Postgres will generate its own index name for child partitions that get added in the future, so it will be pretty difficult to ensure consistent child index names when doing time-based partitioning. I'll push a commit shortly

> One other thing: I think it's OK to require the name to match for the if_not_exists (it saves a lot of effort on trying to lookup indexes and match them opportunistically), but I wonder if we should explicitly document as such.

I believe the base `add_index` Rails method also behaves this way, so since it's consistent I don't think we need explicit documentation